roles/rhsm-subscription/tasks/main.yaml

```diff
@@ -25,15 +26,13 @@
     state: present
     pool: "{{ rhsm_pool }}"
   when: "'Subscribed' not in subscribed.stdout and rhsm_activation_key is defined and rhsm_activation_key != 'None'"
-  register: register_key_result
+  register: register_key_result and ansible_distribution == "RedHat"
```

Comment:
Blocks are awesome. Allowing for the conditionals to be applied in one place is nice, but setting vars at the block-scope is even better :)